File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnClassQuery.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.verbnet.sql;

import android.database.sqlite.SQLiteDatabase;

import org.sqlunet.sql.DBQuery;

/**
 * VerbNet class query
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class VnClassQuery extends DBQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.VerbNetClassQuery;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param classId    target class id
	 */
	@SuppressWarnings("boxing")
	public VnClassQuery(final SQLiteDatabase connection, final long classId)
	{
		super(connection, VnClassQuery.QUERY);
		setParams(classId);
	}

	/**
	 * Get the class id from the result set
	 *
	 * @return the class id from the result set
	 */
	public long getClassId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(0);
	}

	/**
	 * Get the class name from the result set
	 *
	 * @return the class name from the result set
	 */
	public String getClassName()
	{
		assert this.cursor != null;
		return this.cursor.getString(1);
	}

	/**
	 * Get the groupings from the result set
	 *
	 * @return the (|-separated) groupings from the result set
	 */
	public String getGroupings()
	{
		assert this.cursor != null;
		return this.cursor.getString(2);
	}
}

File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnClassQuery.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.verbnet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.sql.DBQuery

/**
 * VerbNet class query
 *
 * @param connection connection
 * @param classId    target class id
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal class VnClassQuery(connection: SQLiteDatabase, classId: Long) : DBQuery(connection, QUERY) {

    init {
        setParams(classId)
    }

    /**
     * Class id from the result set
     */
    val classId: Long
        get() = cursor!!.getLong(0)

    /**
     * Class name from the result set
     */
    val className: String
        get() = cursor!!.getString(1)

    /**
     * (|-separated) Groupings from the result set
     */
    val groupings: String
        get() = cursor!!.getString(2)

    companion object {
        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.VerbNetClassQuery
    }
}
